### ISSUE TITLE
confusing warning when overriding js in theme

### ISSUE THREAD

**Issue by tomaszalusky** [mkdocs/mkdocs#1425]
I need to add extra javascript library to each page.
Unfortunately, the library must be added into head section, otherwise it does not work.
I observed the scripts declared in `pages.yml/extra_javascript` section are added into footer, which is not sufficient for me.

So, according to [this source](http://www.mkdocs.org/user-guide/styling-your-docs/#customizing-a-theme), I put libraries into newly created directory `mkdocs-theme-customization/js`,
overridden `main.html` template with `{% block libs %}` with `<script>` tags
and overridden `theme/custom_dir` to `mkdocs-theme-customization` directory.

The page works correctly. What I don't get is 

```
WARNING -  Config value: 'extra_javascript'. Warning: Some files in your 'docs_dir' are not listed in the 'extra_javascript' config setting and will be ignored by the theme. Add the following files to the 'extra_javascript' config setting if you want them to have an effect on the theme: ['_build/mkdocs-theme-customization/js/flowchart.min.js', '_build/mkdocs-theme-customization/js/raphael.min.js']`
```

This message appears during build when no `extra_javascript` is defined or is empty array.
When I add scripts as error message hints, `<script>` elements are rendered also in footer, i.e. twice in page.
When declaring `extra_javascript: ['anything_nonsense']` invalid `<script src...>` path is rendered, however the warning disappears.

Looks like bug to me. Javascripts which override theme should not be checked for presence in `extra_javascript`.
Or - better - `pages.yml` could support something like `extra_javascript_before` section.

Thanks for taking a look!

**Comment by algo99** [comment]
The official way for this is using a [ custom theme](http://www.mkdocs.org/user-guide/custom-themes/). There you override main.html where you can specify your CSS/JS and they will appear at the top of generated index.html

**Comment by tomaszalusky** [comment]
@algo99 I don't think *fresh new* custom theme is recommended solution to my problem. If you meant *customizing existing* theme, that's what I actually did. What I mind is I get warning message even if I followed manual and this message does not currently make sense to me.

**Comment by waylan** [comment]
@tomaszalusky you are correct that this should be doable with [theme customization][1]. No need to use a custom theme which would replace the entire theme.

[1]: https://www.mkdocs.org/user-guide/styling-your-docs/#using-the-theme-custom_dir

The script files you are listing in the `libs` block of your custom template should be in the `custom_dir`, not in the `docs_dir`. And the `custom_dir` should not be in the `docs_dir`. And the scripts would not need to be listed in the `extra-javascript` setting.  If that's the case and you are still getting the warning, could you provide the relevant parts of your config file (`docs_dir and `theme` settings)?

**Comment by tomaszalusky** [comment]
@waylan Thanks for advice. From the three conditions you stated, the second one was not satisfied in my case - my `custom_dir` was indirect subdirectory of `docs_dir`. After refactoring directory structure to separate sources and configuration, I could entirely remove `extra_javascript` section and the warning disappeared.

I didn't notice anything about this constraint anywhere in manual. If I didn't overlooked, I think the manual could be improved adding your statement **"`custom_dir` should not be in the `docs_dir`"**. Anyway I'm happy with closing this issue.

**Comment by waylan** [comment]
This was addressed in #1376. The updated docs now make it clear that the custom_dir should not be in the docs_dir.
